uat/conftest.py

```diff
@@ -21,8 +25,61 @@ def change_test_dir(tmpdir, monkeypatch):
 @pytest.fixture()
 def gdk_cli(request):
     debug = request.config.getoption("--gdk-debug")
-    print(f"Initializing GDK client with debug mode {debug}")
+    print(f"Initializing GDK client {'in debug mode' if debug else ''}.")
     if request.config.getoption("--instrumented"):
         return GdkInstrumentedProcess(debug)
     else:
         return GdkProcess(debug)
+
+
+@pytest.fixture(autouse=True)
+def version_check(request):
+    version_option = request.config.getoption('--target-version')
+    if version_option is None:
+        return
+
+    latest_mode = version_option == 'HEAD'
+    print(f"\nTarget gdk-cli version {version_option}")
+
+    # request.node is the current test item
+    # query the marker "version" of current test item
+    version_marker = request.node.get_closest_marker('version')
+
+    # if test item was not marked, there's no version restriction
+    if version_marker is None:
+        return
+
+    # parse target version from options
+    target_version = version_parser(version_option) if not latest_mode else version_option
```

Comment:
Why would target_version be either of `version_parser(version_option)` and `version_option`?

Reply:
This simplifies the messages passed to pytest.skip(...) later during the version constraints evaluates.

if its `latest_mode` then `target_version` would evaluate to 'HEAD' string else a `Version` object.